pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -0,0 +1,282 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package nodelabeller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	device_manager "kubevirt.io/kubevirt/pkg/virt-handler/device-manager"
+)
+
+const (
+	oldLabelNamespace              = "feature.node.kubernetes.io"
+	oldLabellerNamespaceAnnotation = "node-labeller-feature.node.kubernetes.io"
+	kvmPath                        = "/dev/kvm"
+	oldCPUFeaturePrefix            = "/cpu-feature-"
+	oldCPUModelPrefix              = "/cpu-model-"
+	oldHyperPrefix                 = "/kvm-info-cap-hyperv-"
+	CPUFeaturePrefix               = "cpu-feature.node.kubevirt.io/"
+	CPUModelPrefix                 = "cpu-model.node.kubevirt.io/"
+	HypervPrefix                   = "hyperv.node.kubevirt.io/"
+)
+
+func NewNodeLabeller(kvmController *device_manager.DeviceController, clusterConfig *virtconfig.ClusterConfig, nodeInformer, kubevirtInformer cache.SharedIndexInformer, clientset kubecli.KubevirtClient, host, namespace string) *NodeLabeller {
+	return &NodeLabeller{
+		kvmController:    kvmController,
+		kubevirtInformer: kubevirtInformer,
+		nodeInformer:     nodeInformer,
+		clientset:        clientset,
+		Queue:            workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+
+		host:           host,
+		namespace:      namespace,
+		logger:         log.DefaultLogger(),
+		clusterConfig:  clusterConfig,
+		virshCommandFN: runVirshCommand,
+		capScanner:     NewCapScanner(),
+	}
+}
+
+//Run runs node-labeller
+func (n *NodeLabeller) Run(threadiness int, stop chan struct{}) {
+	defer n.Queue.ShutDown()
+	n.logger.Infof("node-labeller is running")
+
+	n.cpuInfo.Lock()
```

Comment:
Can you move this lock and unlock to the function that needs them, `loadCPUInfo()`?